### PR TITLE
ConsoleLogRecordExporter modified to support Message and StateValues

### PR DESCRIPTION
**Pull request by cijothomas**
https://github.com/open-telemetry/opentelemetry-dotnet/pull/1869 expanded LogRecord to have formatted message and higher-perf StateValues.
This PR is modifying ConsoleExporter to support these additional fields.

When right options are enabled, the following output is produced.
image

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs [hunk 1]
```diff
@@ -39,7 +39,24 @@ public override ExportResult Export(in Batch<LogRecord> batch)
                 this.WriteLine($"{"LogRecord.CategoryName:".PadRight(rightPaddingLength)}{logRecord.CategoryName}");
                 this.WriteLine($"{"LogRecord.LogLevel:".PadRight(rightPaddingLength)}{logRecord.LogLevel}");
                 this.WriteLine($"{"LogRecord.TraceFlags:".PadRight(rightPaddingLength)}{logRecord.TraceFlags}");
-                this.WriteLine($"{"LogRecord.State:".PadRight(rightPaddingLength)}{logRecord.State}");
+                if (logRecord.Message != null)
+                {
+                    this.WriteLine($"{"LogRecord.Message:".PadRight(rightPaddingLength)}{logRecord.Message}");
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs:44]: Hey, I am wondering if we could output a markdown table, similar like what Benchmark.Net did?

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs [hunk 2]
```diff
@@ -39,7 +39,24 @@ public override ExportResult Export(in Batch<LogRecord> batch)
...
+                if (logRecord.State != null)
+                {
+                    this.WriteLine($"{"LogRecord.State:".PadRight(rightPaddingLength)}{logRecord.State}");
+                }
+                else if (logRecord.StateValues != null)
+                {
+                    this.WriteLine("LogRecord.StateValues (Key:Value):");
+                    for (int i = 0; i < logRecord.StateValues.Count; i++)
+                    {
+                        this.WriteLine($"{logRecord.StateValues[i].Key.PadRight(rightPaddingLength)}:{logRecord.StateValues[i].Value}");
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs:56]: This padding would leave `:` on the right, a bit weird.

Comment (CodeBlanch) [reply src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs:56]: You can see it in the picture in the description, it does look a bit odd.

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs:44]: let me try to get that! would make the getting-started readme looks nicer.

**Pull request merged**